contracts/dispatcher/src/state/mod.rs

```diff
@@ -5,19 +5,16 @@ use sdk::{
     schemars::{self, JsonSchema},
 };
 
-use self::reward_scale::RewardScale;
+use crate::{reward_scale::RewardScale, typedefs::CadenceHours};
 
 pub mod config;
 pub mod dispatch_log;
-pub mod reward_scale;
 
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
 pub struct DispatchLog {
     pub last_dispatch: Timestamp,
 }
 
-pub type CadenceHours = u16;
-
```

Comment:
what is the idea of this refactoring? The structure was good enough ... and since we do not add more code I doubt we need it ... moreover this change is in a wrong direction

Reply:
Again, as I have mentioned before, the logic right now is actually the one in the wrong direction, because we are using types defined in `state`, which is feature gated, in `msg` which isn't.

Reply:
a change in the right direction is one that does not complicate the stuff and the one that really reflects the motivation and usage of the code.

So, a quick look at the code reveals that the feature "contract" is totally useless because there is no valid usage of the package as a "non-contract". Therefore a simple change that eliminates that feature is a perfect fit.